commands/roles.js

```diff
@@ -0,0 +1,22 @@
+const { roleBlacklist } = require('../config.json');
+
+module.exports = {
+	name: 'roles',
+	description: 'Lists all self-assignable roles.',
+	aliases: [],
+	usage: '[roles]',
+	cooldown: 5,
+	guildOnly: true,
+	execute(message, args) {
+
+        let assignableRoles = message.guild.roles.values()
+            .filter(role=>!roleBlacklist.includes(role.name))
```

Comment:
We can't `filter` this, as `message.guild.roles.values()` returns an iterator object, not an array. This is because it is a Map (it's actually a [Collection](https://discord.js.org/#/docs/main/stable/class/Collection) as per the docs), which is why we can do `.values()` in the first place.

The best way to return a list of roles by name is just to create an array instead.
```js
let roles = [];
for (let role of message.guild.roles.values()) {
  roles.push(role.name);
}
```

Reply:
Surely using `roles =`[`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)`(message.guild.roles.values())` would be a little more concise 👍

Reply:
That's perfect!